### ISSUE TITLE
java.lang.SecurityException: !@Too many alarms (500)

### ISSUE THREAD

**Issue by anukools** [OneSignal/OneSignal-Android-SDK#168]
These are the logs that I got from fabric showing crash  after 

```Java
Fatal Exception: java.lang.RuntimeException: Unable to start service com.onesignal.SyncService@837f9bc with Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=in.inspirelabs.grabon/.app.SplashActivity bnds=[172,532][297,675] (has extras) }: java.lang.SecurityException: !@Too many alarms (500) registered from pid 13675 uid 10233
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4145)
       at android.app.ActivityThread.access$2400(ActivityThread.java:229)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1924)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:7325)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by java.lang.SecurityException: !@Too many alarms (500) registered from pid 13675 uid 10233
       at android.os.Parcel.readException(Parcel.java:1620)
       at android.os.Parcel.readException(Parcel.java:1573)
       at android.app.IAlarmManager$Stub$Proxy.set(IAlarmManager.java:217)
       at android.app.AlarmManager.setImpl(AlarmManager.java:484)
       at android.app.AlarmManager.set(AlarmManager.java:260)
       at com.onesignal.SyncService.scheduleServiceRestart(SourceFile:124)
       at com.onesignal.SyncService.onTaskRemoved(SourceFile:100)
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4130)
       at android.app.ActivityThread.access$2400(ActivityThread.java:229)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1924)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:7325)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

**Comment by jkasten2** [comment]
Thanks for the report, we will try reproducing this.

**Comment by jkasten2** [comment]
@anukools This error seems to only happen on specific Samsung devices. I have fixed this in commit b26fa8d1d5922151bb1e7616200c22ddb16c5888.

It will be included in version 3.4.4 of our SDK. Please reply back confirming the fix once it is released if you can.

Thanks.
